update container labels docs with complete table of labels provided by the tooling (#44638)
dotnet/docs: docs/core/containers/publish-configuration.md

### Default container labels

> Labels are often used to provide consistent metadata on container images. This package provides some default labels to encourage better maintainability of the generated images. All default label generation can be disabled by setting `ContainerGenerateLabels` to `false`. In addition, each default label has an  individual enablement flag that can be set to `false` to disable that specific label.

Reviewer: > This package provides...

Which package?

Author: Ah, bad verbiage from when the container tools were primarily delivered by the Microsoft.Net.Build.Containers NuGet package. Will patch this up to remove the reference.

> Where possible, we try to make use of existing MSBuild properties to provide the values for these labels, though we also provide properties that allow for explicit control of the values.

Reviewer: ```suggestion
Where possible, existing MSBuild properties provide the values for these labels. Other properties allow for explicit control of their values.
```

Reviewer: Style guidance is to avoid we/our.

| Annotation                                                                           | Default Value                                                                                      | Dedicated Property Name      | Fallback Property Name     | Enabled Property Name                       | Notes                                                                                                               |
| ------------------------------------------------------------------------------------ | -------------------------------------------------------------------------------------------------- | ---------------------------- | -------------------------- | ------------------------------------------- | ------------------------------------------------------------------------------------------------------------------- |

> | `org.opencontainers.image.created` and `org.opencontainers.artifact.created`         | the [RFC 3339](https://tools.ietf.org/html/rfc3339#section-5.6) format of the current UTC DateTime |                              |                            | `ContainerGenerateLabelsImageCreated`       |                                                                                                                     |

Reviewer: ```suggestion
| `org.opencontainers.image.created` and `org.opencontainers.artifact.created`         | The [RFC 3339](https://tools.ietf.org/html/rfc3339#section-5.6) format of the current UTC DateTime |                              |                            | `ContainerGenerateLabelsImageCreated`       |                                                                                                                     |
```
